e2e/cypress/integration/search/post_search_display_spec.js

```diff
@@ -188,7 +196,267 @@ describe('Post search display', () => {
+
+            // GM case with @mention
+            cy.apiCreateGroupChannel(groupMembers.map((member) => member.id)).then(({channel}) => {
+                cy.visit(`/${testTeam.name}/messages/${channel.name}`);
+
+                const message = `Group message @${testUser.username}`;
+
+                // # Post a GM message
+                cy.postMessage(message);
+            });
+        });
+
+        // Public channel @channel case
+        cy.apiCreateChannel(testTeam.id, 'channel-mentions', 'channel-mentions').then(({channel}) => {
+            cy.postMessageAs({sender: adminUser, message: '@channel', channelId: channel.id});
+        });
+
+        // Off-Topic with @mention
+        cy.apiGetChannelByName(testTeam.name, 'Off-Topic').then(({channel}) => {
+            // # Have another user send a post
+            cy.postMessageAs({sender: adminUser, message: `@${testUser.username} greetings`, channelId: channel.id});
+        });
+
+        // DM case
+        cy.apiCreateDirectChannel([testUser.id, adminUser.id]).then(({channel}) => {
+            // # Send a DM with @mention
+            cy.postMessageAs({sender: adminUser, message: `@${testUser.username} direct`, channelId: channel.id});
+        });
+
+        cy.apiLogin(testUser).then(() => {
+            // # Visit town square as an admin
+            cy.visit(`/${testTeam.name}/channels/town-square`);
+
+            // # Click the @ mention filter on search
+            cy.get('#channelHeaderMentionButton').click();
+
+            // * RHS should be visible with search results
+            cy.get('#search-items-container').should('be.visible');
+
+            // * Assert that results include all messages apart from the @channel one
+            cy.get('.search-item__container .post.post--thread .post__content .post__body').each(($item) => {
+                expect($item.text()).to.match(new RegExp(`@${testUser.username} (direct|greetings)|Group message @${testUser.username}`));
+            });
+        });
+    });
+
+    it('MM-T2299 - Search results include matches in DMs', () => {
+        cy.apiCreateDirectChannel([testUser.id, adminUser.id]).then(({channel}) => {
+            // # Send a DM to another user
+            cy.postMessageAs({sender: adminUser, message: 'This is direct', channelId: channel.id});
+        });
+
+        cy.apiLogin(testUser).then(() => {
+            // # Visit town-square
+            cy.visitAndWait(`/${testTeam.name}/channels/town-square`);
+
+            // # Search for direct
+            cy.get('input#searchBox').click().type('direct{enter}').wait(TIMEOUTS.HALF_SEC);
+
+            // * RHS should be visible with search results
+            cy.get('#search-items-container').should('be.visible');
+
+            // * Assert that results include only the verify message
+            cy.get('.search-item__container .post.post--thread').each(($item) => {
+                cy.wrap($item).should('contain.text', 'This is direct');
```

Comment:
Failed here since searching for "direct" term returned two posts. The first one is posted from MM-T2298.

<img width="1599" alt="Screen Shot 2021-02-04 at 5 12 59 PM" src="https://user-images.githubusercontent.com/5334504/106870809-7e3b0e00-670c-11eb-920f-d0408576af29.png">